codec-http/src/main/java/io/netty/handler/codec/http/multipart/AbstractMemoryHttpData.java

```diff
@@ -43,6 +43,7 @@
 
     protected AbstractMemoryHttpData(String name, Charset charset, long size) {
         super(name, charset, size);
+        byteBuf = EMPTY_BUFFER;
```

Comment:
Why this change? There are a number of places in the code where we check if `byteBuf` is `null`. Also, when `delete` is called, the `byteBuf` is set to `null`.

Reply:
The reason is the different behavior between Memory and Disk. When we allocate a Disk based, the buffer, not here, is like it is there. In Memory, it was not.
Therefore, for those realying on existing buffer, even empty, is not correct (as reporting by the user in his issue). He used the `decoder.currentPartialHttpData()` in his code (legal), but then when he asked for the buffer, he's got "null" instead of an empty buffer when the buffer has not yet data pushed by the decoder.

When the buffer is released, the buffer is set to null after releasing it.
So it does not change the global behavior but makes both implementation seemless the same, which was the main issue.

Note that the change also gives more chance to use the current decoded buffer to fill the `currentHttpData`.

Reply:
Shall be kept